### ISSUE TITLE
libxml2/host: build failures

### ISSUE THREAD

**Issue by micmac1** [openwrt/packages#6073]
Maintainer: @mhei 
Environment: all

Description:
The build of libxml2/host fails sporadically on the build bots with:

```
make[6]: Entering directory '/build/openwrt-18.06/slaves/phase2/mipsel_24kc/build/sdk/build_dir/hostpkg/libxml2-2.9.8'
  CC       SAX.lo
In file included from /usr/include/stdlib.h:42:0,
                 from ./include/libxml/SAX.h:16,
                 from ./include/libxml/globals.h:20,
                 from ./include/libxml/threads.h:35,
                 from ./include/libxml/xmlmemory.h:218,
                 from ./include/libxml/tree.h:1307,
                 from ./include/libxml/c14n.h:27,
                 from elfgcchack.h:17,
                 from libxml.h:118,
                 from SAX.c:12:
/usr/include/x86_64-linux-gnu/bits/waitstatus.h:79:15: error: duplicate member '__w_retcode'
  unsigned int __w_retcode:8;
               ^
/usr/include/x86_64-linux-gnu/bits/waitstatus.h:80:15: error: duplicate member '__w_coredump'
  unsigned int __w_coredump:1;
               ^
/usr/include/x86_64-linux-gnu/bits/waitstatus.h:81:15: error: duplicate member '__w_termsig'
  unsigned int __w_termsig:7;
               ^
/usr/include/x86_64-linux-gnu/bits/waitstatus.h:93:15: error: duplicate member '__w_stopsig'
  unsigned int __w_stopsig:8; /* Stopping signal.  */
               ^
/usr/include/x86_64-linux-gnu/bits/waitstatus.h:94:15: error: duplicate member '__w_stopval'
  unsigned int __w_stopval:8; /* W_STOPPED if stopped.  */
               ^
Makefile:1248: recipe for target 'SAX.lo' failed
```

On the build bots running the openwrt-18.06 snapshot builds this always fails. I suppose because just one machine is used, or identical machines.

I cannot reproduce this on my own boxes, so I'm not sure what the problem really is. Maybe anybody else knows.

This causes follow-up failures for asterisk, so would be appreciated if this could get fixed.

Best regards,
Seb

**Comment by micmac1** [comment]
The union wait code was removed from glibc in commit b49ab5f4503f36dcbf43f821f817da66b2931fe6. So that removed all the members visible in the errors (e.g. '__w_retcode'). I checked my distro's current glibc and it doesn't have the code anymore.

I installed Ubuntu 18.04 server with build-essential and libncurses5-dev. It doesn't have the code neither. Compiled libxml2/host in the SDK, no problem.

Went back further, installed Ubuntu 16.04 Server. Same procedure. All fine. It still has the union wait code.

I suppose there's a chance there's something wrong with the setup of the builder. According to [this overview](http://release-builds.lede-project.org/18.06/packages/builders) that would be openwrt-18.06-phase2. Any chance the setup could be checked?
